libpromises/attributes.c

```diff
@@ -1489,6 +1489,17 @@ EditRegion GetRegionConstraints(const EvalContext *ctx, const Promise *pp)
     e.select_end = PromiseGetConstraintAsRval(pp, "select_end", RVAL_TYPE_SCALAR);
     e.include_start = PromiseGetConstraintAsBoolean(ctx, "include_start_delimiter", pp);
     e.include_end = PromiseGetConstraintAsBoolean(ctx, "include_end_delimiter", pp);
+    char *value = PromiseGetConstraintAsRval(pp, "select_end_match_eof", RVAL_TYPE_SCALAR);
+    if (value && (strcmp(value, "false") == 0))
+    {
+        e.select_end_match_eof = false;
+    }
+    else
+    {
+        /* Default value for 'select_end_match_eof' is true. */
+        e.select_end_match_eof = true;
```

Comment:
It seems like the consensus in [CFE-2263](https://tracker.mender.io/browse/CFE-2263) is to have this be false by default.